### PR TITLE
core-lightning: restart bitcoind when we're restarted.

### PR DESCRIPTION
**Pull request by rustyrussell**
Nobody else seems to, which explains some of our issues?

No, still failing.  Let me add a crapton of logging, since it only fails on GH!

Fixes https://github.com/rustyrussell/lnprototest/issues/31

### CODE REVIEW

File: lnprototest/backend/bitcoind.py [hunk 1]
```diff
@@ -124,6 +123,10 @@ def __is__bitcoind_ready(self) -> bool:
         return True
 
     def start(self) -> None:
+        # Now we're about to start, we can allocate the port.
+        # We used to do this at init, but in parallel cases another
+        # bitcoin could use the port when we restart!
+        self.port = reserve()
```
Comment (vincenzopalazzo) [inline lnprototest/backend/bitcoind.py:129]: Good catch!

**Comment by vincenzopalazzo** [comment]
The only regression that we have with this PR is the execution time of the test, In particular:

- #37 runs in 12m 0s
- #52 runs in 56m 30s (and still running)

I will check this in local and see what is going on!

**Comment by rustyrussell** [comment]
See final fixup commit!  If we don't rewrite the bitcoin.conf when we change port, it all breaks...

**Comment by vincenzopalazzo** [comment]
Closing in favor of https://github.com/rustyrussell/lnprototest/pull/53 that includes a patch of this patch.

In addition, the CI now run @niftynei cln patch https://github.com/ElementsProject/lightning/pull/5367 so we start to monitoring other CI failure, but at the moment this problem looks fixed

**Pull request closed without merging**